packages/publikator/graphql/resolvers/_mutations/commit.js

```diff
@@ -28,33 +29,61 @@ const {
   }
 } = require('@orbiting/backend-modules-documents')
 
+const {
+  ASSETS_SERVER_BASE_URL
+} = process.env
+
+const generateImageData = async (blob) => {
+  const meta = await sharp(blob).metadata()
+  // image/jpeg -> jpeg
+  // image/png -> png
+  // image/svg+xml -> svg
+  const suffix = blob.type.split('/')[1].split('+')[0]
+  const hash = hashObject(blob)
+  const path = `images/${hash}.${suffix}`
+  const url = `${path}?size=${meta.width}x${meta.height}`
+  return {
+    image: {
+      path,
+      hash,
+      blob
+    },
+    url
+  }
+}
+
 const extractImage = async (url, images) => {
   if (url) {
     let blob
     try {
       blob = dataUriToBuffer(url)
     } catch (e) { /* console.log('ignoring image node with url:' + url) */ }
     if (blob) {
-      const meta = await sharp(blob).metadata()
-      // image/jpeg -> jpeg
-      // image/png -> png
-      // image/svg+xml -> svg
-      const suffix = blob.type.split('/')[1].split('+')[0]
-      const hash = hashObject(blob)
-      const path = `images/${hash}.${suffix}`
-      const url = `${path}?size=${meta.width}x${meta.height}`
-      const image = {
-        path,
-        hash,
-        blob
-      }
+      const { image, url } = await generateImageData(blob)
       images.push(image)
       return url
     }
   }
   return url
 }
 
+const isFromRepo = (url, repoId) =>
+  url && url.startsWith(`${ASSETS_SERVER_BASE_URL}/github/`) && !url.includes(repoId)
+
+const importFromRepo = async (url, images, repoId) => {
+  if (isFromRepo(url, repoId)) {
+    try {
+      const blob = await fetch(url).then(result => result.buffer())
+      const { image, newUrl } = await generateImageData(blob)
+      images.push(image)
+      return newUrl
+    } catch (e) {
+      console.error('failed to transfer image', e)
```

Comment:
will help us if we want to case it down:
```suggestion
      console.error('failed to transfer image', repoId, url, e)
```